x-pack/plugins/maps/public/lens/choropleth_chart/expression_function.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExpressionFunctionDefinition } from 'src/plugins/expressions/common';
+import type { LensMultiTable } from '../../../../lens/common';
+import type { ChoroplethChartConfig, ChoroplethChartProps } from './types';
+import { RENDERER_ID } from './expression_renderer';
+
+interface ChoroplethChartRender {
+  type: 'render';
+  as: 'lens_choropleth_chart_renderer';
+  value: ChoroplethChartProps;
+}
+
+export const getExpressionFunction = (): ExpressionFunctionDefinition<
+  'lens_choropleth_chart',
+  LensMultiTable,
+  Omit<ChoroplethChartConfig, 'layerType'>,
+  ChoroplethChartRender
+> => ({
+  name: 'lens_choropleth_chart',
+  type: 'render',
+  help: 'A choropleth chart. Metrics are joined to vector features to compare values across political boundaries',
+  args: {
+    title: {
+      types: ['string'],
+      help: '',
+    },
+    description: {
+      types: ['string'],
+      help: '',
+    },
+    layerId: {
+      types: ['string'],
+      help: '',
+    },
+    emsField: {
+      types: ['string'],
+      help: 'Elastic Map Service boundaries layer field provides the vector feature join key',
+    },
+    emsLayerId: {
+      types: ['string'],
+      help: 'Elastic Map Service boundaries layer id that provides vector features',
+    },
+    regionAccessor: {
+      types: ['string'],
+      help: 'Bucket accessor identifies the region key column',
+    },
+    valueAccessor: {
+      types: ['string'],
+      help: 'Value accessor identifies the value column',
+    },
+  },
+  inputTypes: ['lens_multitable'],
+  fn(data, args) {
+    return {
```

Comment:
This is a very recent change but we moved the responsibility of logging the data table to the visualization function. Could you add the same integration as xy chart here ?https://github.com/elastic/kibana/blob/11451adb63f0e342b29c90db170ebc1324fa43cf/x-pack/plugins/lens/common/expressions/xy_chart/xy_chart.ts#L161

Reply:
I had to revert these changes. logDataTable is not exposed to plugins. I created https://github.com/elastic/kibana/issues/127995 to track. 

@flash1293 and I chatted offline and @flash1293 is ok with merging this PR without logging data table in the visualization function. This can be addressed in https://github.com/elastic/kibana/issues/127995